src/extensions/default/CodeFolding/Prefs.js

```diff
@@ -0,0 +1,116 @@
+                           {name: strings.FADE_FOLD_BUTTONS, description: strings.FADE_FOLD_BUTTONS_HELP});
+    prefs.definePreference("maxFoldLevel", "number", 2,
+                           {name: strings.MAX_FOLD_LEVEL, description: strings.MAX_FOLD_LEVEL_HELP});
+    prefs.definePreference("folds", "object", {});
+
+    /**
+      * Simplifies the fold ranges into an array of pairs of numbers.
+      * @param {!Object} folds the raw fold ranges indexed by line numbers
+      * @return {Object} an object whose keys are line numbers and the values are array
+      * of two 2-element arrays. First array contains [from.line, from.ch] and the second contains [to.line, to.ch]
+      */
+    function simplify(folds) {
+        if (!folds) {
+            return;
+        }
+        var res = {}, range;
+        Object.keys(folds).forEach(function (line) {
+            range = folds[line];
+            res[line] = Array.isArray(range) ? range : [[range.from.line, range.from.ch], [range.to.line, range.to.ch]];
+        });
+        return res;
+    }
+
+    /**
+      * Inflates the fold ranges stored as simplified numeric arrays. The inflation converts the data into
+      * objects whose keys are line numbers and whose values are objects in the format {from: {line, ch}, to: {line, ch}}.
+      * @param {Object}  folds the simplified fold ranges
+      * @return {Object} the converted fold ranges
+      */
+    function inflate(folds) {
+        if (!folds) {
+            return;
+        }
+         //transform the folds into objects with from and to properties
+        var ranges = {}, obj;
+        Object.keys(folds).forEach(function (line) {
+            obj = folds[line];
+            ranges[line] = {from: {line: obj[0][0], ch: obj[0][1]}, to: {line: obj[1][0], ch: obj[1][1]}};
+        });
+
+        return ranges;
+    }
+
+    /**
+      * Gets the line folds saved for the specified path.
+      * @param {string} path the document path
+      * @return {Object} the line folds for the document at the specified path
+      */
+    function getFolds(path) {
+        store = (prefs.get(foldsKey) || {});
+        return inflate(store[path]);
+    }
+
+    /**
+      * Saves the line folds for the specified path
+      * @param {!string} path the path to the document
+      * @param {Object} folds the fold ranges to save for the current document
+      */
+    function setFolds(path, folds) {
+        store[path] = simplify(folds);
+        prefs.set(foldsKey, store);
+    }
+
+    /**
+      * Get the code folding setting with the specified key from the store
+      * @param {!string} key The key for the setting to retrieve
+      * @return {string} the setting with the specified key
+      */
+    function getSetting(key) {
+        return prefs.get(key);
+    }
+
+    /**
+      * Clears all the saved line folds for all documents.
+      */
+    function clearAllFolds() {
+        prefs.set(foldsKey, {});
```

Comment:
This function appears to be unused, but I think it also has a bug: this doesn't reset the `store` var used elsewhere, so any call to `setFolds()` later seems like it would immediately restore _all_ the folds that had been cleared previously.

It's probably safest just to re-get the pref before setting it each time.  There may be other bugs lurking with the current approach too (e.g. if you edit the prefs file... does it blow those changes away?)